pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -37,6 +37,12 @@ const (
 	ReasonConditionCheckFailed = "ConditionCheckFailed"
 )
 
+var (
+	// ErrInvalidTaskResultReference indicates that the reason for the failure status is that there
+	// is an invalid task result reference
+	ErrInvalidTaskResultReference = errors.New("Invalid task result reference")
```

Comment:
nit: this constant is declared in this file but only used in resultrefresolution.go. Consider declaring it there